arho_feature_template/core/feature_template.py

```diff
@@ -0,0 +1,31 @@
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import TYPE_CHECKING, Sequence
+
+if TYPE_CHECKING:
+    from qgis.core import QgsMapLayer, QgsVectorLayer
+    from qgis.gui import QgsDoubleSpinBox, QgsSpinBox
+    from qgis.PyQt.QtGui import QIcon
+    from qgis.PyQt.QtWidgets import QLineEdit
+
+
+@dataclass
+class FeatureAttribute:
+    name: str
+    display_name: str
+    feature_attribte_group: str
+    attribute_layer: QgsMapLayer  # Is this correct type?
+    additional_information: str
+    input_field_type: QLineEdit | QgsSpinBox | QgsDoubleSpinBox | None
+    modifiable: bool = False
+    hidden: bool = False
+
+
+@dataclass
+class FeatureTemplate:
```

Comment:
This is now missing possible child features. But that's totally fine for now! Let's first try to get this working with the most simple one feature case. Probably the library config should be updated accordingly.